core/domain/classifier_services.py

```diff
@@ -127,6 +127,69 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
```

Comment:
Yes, because you have no fixed bound here. It's important to think also of the worst-case scenario, and there may be explorations where the states are mostly text.

Also, it's not that hard to redesign, so I think we should just do it correctly with regards to performance from the get-go.